test/pdo_sqlsrv/pdo_378_out_param_error.phpt

```diff
@@ -0,0 +1,60 @@
+--TEST--
+This test verifies that GitHub issue #378 is fixed in pdo_sqlsrv.
+--DESCRIPTION--
+GitHub issue #378 - output parameters appends garbage info when variable is initialized with different data type
+steps to reproduce the issue:
+1- create a store procedure with print and output parameter
+2- initialize output parameters to a different data type other than the type declared in sp.
+3 - call sp.
+--FILE--
+<?php
+require_once("pdo_tools.inc");
+require_once("autonomous_setup.php");
+
+$conn = new PDO( "sqlsrv:Server=$serverName; Database = tempdb ", $username, $password);
+if (!$conn) {
+	print_r($conn->errorInfo());
+}
+
+//----------------Main---------------------------
+$procName = GetTempProcName();
+createSP($conn, $procName);
+executeSP($conn, $procName);
+$conn->setAttribute( PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION );
+executeSP($conn, $procName);
+echo "Done";
+
+//-------------------functions-------------------
+function createSP($conn, $procName){
+	
+	$sp_sql="create proc $procName @p1 integer, @p2 integer, @p3 integer output
+	as
+	begin
+		select @p3 = @p1 + @p2
+		print @p3
+	end
+	";
+	$stmt = $conn->exec($sp_sql);
+	if ($stmt === false) { print("Failed to create stored procedure"); }
+}
+
+function executeSP($conn, $procName){
+	$expected = 3;
+	$stmt = $conn->prepare("{call $procName( ?, ?, ? )}");
+	$stmt->bindParam(1, $v1);
+	$stmt->bindParam(2, $v2);
+	$stmt->bindParam(3, $v3, PDO::PARAM_INT, 10);
```

Comment:
Yeah, the length should be set, otherwise driver won't consider it as output param